### ISSUE TITLE
INLINE_SOURCES = YES  do not generate code inline from sources if there is no @return after @brief. 

### ISSUE THREAD

**Issue by ikvasir** [doxygen/doxygen#8585]
**Describe the bug**
After checking `INLINE_SOURCES = YES` in html there is no source code inserts if i just write `///@brief`. But after adding `///@return`  it works. 

**Expected behavior**
```
//first version
	///@brief function without return - this will not be inserted inline
	void func()
	{
		1 + 2;
		3 << 4;
		5;
	};
///second version
	///@brief function with return - this will insert
	///@return void
	void func2()
	{
		1 + 2;
		3 << 4;
		5;
	};
```

After reading manual I `v expected first version because: "YES will include the body of functions". But works only second version. with `///@return`

**Screenshots**
![maybe_a_bug](https://user-images.githubusercontent.com/10597377/120507855-da1e5680-c3cf-11eb-9b8a-7423839d42dd.png)

**Version**
Window 10 64 bit - Doxygen 1.9.1 installed from [here](https://doxygen.nl/files/doxygen-1.9.1-setup.exe)

**Comment by albert-github** [comment]
The problem here is that `func1` doesn't have any detailed description, but `func2` has (in the form of the `return`, though the line `/// second version` is a bit disturbing the picture as well.
As there is no detailed description there is no place to place the inline source. It is possible to enforce  a detailed section by means of: `ALWAYS_DETAILED_SEC    = YES`

@doxygen should as an enhancement:  in case `INLINE_SOURCES = YES` and when there are inline sources present automatically add a detailed section?

Furthermore, the source code is present but not complete, please always:
- attach a, small, self contained example (source+configuration file in a tar or zip) that allows us to reproduce the problem? Please don't add external links as they might not be persistent.

**Comment by ikvasir** [comment]
Here  config file. 
[Doxyfile.zip](https://github.com/doxygen/doxygen/files/6587730/Doxyfile.zip)
All source in question. 
And `ALWAYS_DETAILED_SEC` was `NO`, and after set it to `YES` both version of functions generate inline code. Thank you.

So it's not bug, it's a feature? I'm new at doxygen and when i find that doxygen do not generate inline source after set `INLINE_SOURCES = YES` i began to read documents hard and than stackoverflow and than i did not find any information that to generate inline sources the detailed section must be present.  And actually what the reason not to generate inline sources if there is no detailed description? For what task may it be necessary not to generate an inline source if the function has only brief but not details? I think this behavior of the program confuses the person who has read the docs first time.  If the function was not important i will never see this. But there is a solution to this, which mean this is not a bug, I close the question. 
So may be there is enough reason to change the INLINE_SOURCES description. Now it  is`Setting the INLINE_SOURCES tag to YES will include the body of functions, classes and enums directly into the documentation. `  i suggest changing to `Setting the INLINE_SOURCES tag to YES will include the body of functions with detailed section, classes and enums directly into the documentation. `

**Comment by albert-github** [comment]
- I've reopened the issue seen my remark with my previous comment, I think it might be handled as an enhancement.
- you suggested some new text, although better I don't the reader will see the subtle different part / will be able to interpret it correctly:  `with detailed section`

**Comment by ikvasir** [comment]
I'v extend the test to clarify how setting `INLINE_SOURCES` to `YES` work and thats the result.
Source, config file and generated html [here](https://github.com/doxygen/doxygen/files/6591437/INLINE_SOURCES_TEST.zip)

Except what is written above:
1) The enum, and enum class in c++ have the same behaviour like function. Without '@details' source was not inserted inline.
2) Class do not insert source code inline anyway without or with '@details' for member functions it only show list of members of class but not code of members. In general it reasonable if i whant to add only class overview without code everithing is ok, but how to add full class source inline? After all, this is what i expected after read the `INLINE_SOURCES`  description 'inline include body of .. clases'.  But now it include not body, but names of methods and variables. If i set `EXTRACT_ALL = YES` it works. In test it ok but in real project i want to hide many of entities and show just may be full code of some documented classes inline.
3) Lambda functions inline source generating not correct, it is look like variables with init section not like functions with inline source. This is slightly different topic.

Of course, I can get around all these limitations using the existing tools like add source between `@code` - `@endcode` or using enabled sections, or use `@snippet`. But i'm to lazy to do that so, here i want to add some proposal to extend inline sources functionality.
If `INLINE_SOURCES = YES` two new flags become availible.
1) FORCE_INCLUDE_STRUCTURES_INLINE_SOURCES 
If set YES overrides any other behaviour(except it must be documented) of including inline sources for classes, structures and unions. And add source code inline for every doxygen documented structures(struct,class,unions).
Default set to NO, so for existing project nothing changed.
2) FORCE_INCLUDE_METHODS_ENUMS_INLINE_SOURCES 
The same for functions, enums, lambdas.

Or may be instead of new flags and using `@cond` - `@endcond` or enabled section for this task, it will be added special command like `@__inlinesource` which place full source of any entity method, enum or struct to docs. But it really looks like `@snippet` works.
So may be the best of all in this way to just override `@snippet` wihout args which add inline source code for current documented entity using force_include:
```
///@brief func
///@snippet or ///@__inlinesource
void f(){ /*code*/} ;

///@brief class
///@snippet or ///@__inlinesource
class C{ /*code*/;} 

```

**Comment by albert-github** [comment]
I don't think it is necessary to introduce new settings for these special cases, with the option suggested in https://github.com/doxygen/doxygen/issues/8585#issuecomment-853149548 I think the mentioned items will also get their automatic detailed section.
Furthermore we are not so happy to introduce new settings as there are already al lot of settings.

**Comment by doxygen** [comment]
> @doxygen should as an enhancement: in case `INLINE_SOURCES = YES` and when there are inline sources present automatically add a detailed section? i.e. add in `memberdef.cpp` function: `MemberDefImpl::isDetailedSectionLinkable()`:
> 
> ```
>   static bool inlineSources = Config_getBool(INLINE_SOURCES);
>   bool docFilter = ....
>  (inlineSources && hasSources())
>  ...
> ```

Yes indeed, that seems to be missing now.

**Comment by albert-github** [comment]
I did some tests already and it looks like to give some good results.
Probably this should be further extended to things like call/ caller graph, reference(by)  relation , reimplements as well.
For the brief description top of the page) there are also some extensions necessary
- the "More..." looks to be easy to accomplishe (according to my test)
- the linking of the the member name / function name now when there is no detailed description this is just the name without a link (I need to dive into this a little bit further).

**Comment by albert-github** [comment]
I've just pushed a proposed patch, pull request #8592

**Comment by albert-github** [comment]
Code has been integrated in master on github (please don't close the issue as this will be done at the moment of an official release).

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.9.2.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
